adal/cache_driver.py

```diff
@@ -164,11 +164,13 @@ def _refresh_entry_if_necessary(self, entry, is_resource_specific):
         now_plus_buffer = now + timedelta(minutes=Misc.CLOCK_BUFFER)
 
         if is_resource_specific and now_plus_buffer > expiry_date:
-            self._log.info('Cached token is expired.  Refreshing: %s', expiry_date)
-            return self._refresh_expired_entry(entry)
+            if TokenResponseFields.REFRESH_TOKEN in entry:
```

Comment:
Good point. I'll remove the unused cache entry in my next commit. A side note: the current cache logic is somewhat complicated. We may refactor them someday in future, probably starting from scratch in MSAL implementation.